http/retryhttp.go

```diff
@@ -0,0 +1,25 @@
+package http
+
+import (
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+// NewHTTPClient returns a retryable HTTP client with common defaults
+func NewHTTPClient(logger log.Logger) *retryablehttp.Client {
```

Comment:
Clients will use this function this way: `http.NewHTTPClient`, while earlier this was: `retry.NewHTTPClient`. I prefer the older syntax, as it gives more context to this package (**retry**).

What about using `httpretry` as package name (-> `httpretry.NewClient`) or using the original package name (`retry` -> `retry.NewHTTPClient`)?